module1/homework/grayscaleImages/test.cpp

```diff
@@ -3,7 +3,7 @@
 #include <utility>  // for std::pair<>
 #include <vector>
 
-// TODO: include
+#include "compression.hpp"
```

Comment:
You could remove vector here since it's in the header.

Reply:
Imagine i don't know what is inside compression header.  In that case I prefer to leave includes used in working file.

Reply:
But here you know what is inside :)

Reply:
And you will usually always have access to header.

Reply:
And what about include-what-you-use?

Reply:
> 
> 
> And you will usually always have access to header.

But what about IWYU?

Reply:
But what about include-what-you-use?

Reply:
When you include header you include all what's in it. The whole point of header is that you have there libraries you use everywhere(so you do not repeat yourself), defenition of functions, aliases, constexprs.Things that usually would be at beginning of the program. By extracting them you make program more readable and in case your projects has more files you do not need to copy "definitions" everywhere.

Reply:
Sorry. My mistake i wasn't aware of "Include what you use rule".